Module-2-Basic-Programming/Day-1/Tasks/Mujaheed.task

```diff
@@ -0,0 +1,52 @@
+1. Explain about variable hoistiog with reasoning ?
+
+A. 1. Declarations are hoisted, not initializations.
+   2. Allows calling functions before their declarations.
+   3. All variable and function declarations are processed before any code execution.
+   4. Undeclared variables are implicitly created as global variables when assigned a value.
+
+    Example Program:
+
+    function codejs(){
+    a = 10;
+    let b = 50;                           
+    }
+    codejs();
+   console.log(a);  
+   console.log(b);   
+
+   <!-- Here the variable hoisting is access these variables
+        and function even before you have initialized it or you have
+        put some value in it. you can access it without any error-->  
+
+2. Explain about type casting by reasoning with examples ?
+
+A. Type conversion (or typecasting) means transfer of data from one data type to another.
+
+    -> "5" + 5 = '55'
```

Comment:
Justification on how the casting is being done here is not given for each statement.